lib/api/api_test.go

```diff
@@ -745,6 +745,19 @@ func TestCSRFRequired(t *testing.T) {
 	if resp.StatusCode != http.StatusOK {
 		t.Fatal("Getting /rest/system/config with API key should succeed, not", resp.Status)
 	}
+
+	// Calling on /rest with the API key as a bearer token should succeed
```

Comment:
Should have tests for wrong API keys?  As this changes what is accepted, we should make sure nothing creeps in later that accepts too much.

Reply:
Added, and also improved & modernized the tests a little